// @devmaxime/capacitor-health-connect/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

const HealthConnect = core.registerPlugin('HealthConnect', {
    web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.HealthConnectWeb()),
});

class HealthConnectWeb extends core.WebPlugin {
    async checkAvailability() {
        console.warn('HealthConnect is not available on the web');
        return { availability: 'NotSupported' };
    }
    async requestPermissions(options) {
        console.warn('HealthConnect is not available on the web', options);
        return { read: [], write: [] };
    }
    async getGrantedPermissions() {
        console.warn('HealthConnect is not available on the web');
        return { read: [], write: [] };
    }
    async revokePermissions() {
        console.warn('HealthConnect is not available on the web');
    }
    async readRecords(options) {
        console.warn('HealthConnect is not available on the web', options);
    }
    async aggregateRecords(options) {
        console.warn('HealthConnect is not available on the web', options);
        return { aggregates: [] };
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    HealthConnectWeb: HealthConnectWeb
});

exports.HealthConnect = HealthConnect;
